accepted/0000-cve-auditing-with-oval.md

```diff
@@ -0,0 +1,188 @@
+
+```mermaid
+flowchart TD
+    A[Get the list of affected products] --> B{Is system product\n affected?}
+    B -->|Yes, product affected| G{Evaluate CVE\n vulnerability criteria tree}
+    B -->|No, product not affected| F[STATUS: NOT_AFFECTED]
+    G --> |Evaluation: true| H[Collect the list of patched\n and affected packages from\n the vulnerability definiton]
+    H --> J{Remove from the list\n the packages that have \na patch in the assigned\n channels and check\n list again}
+    J --> | List is empty| K[STATUS: AFFECTED_FULL_PATCH_APPLICABLE]
+    J --> | List size decreased| M[STATUS: AFFECTED_PARTIAL_PATCH_APPLICABLE]
+    J --> | Else, list not empty and haven't changed | L{"One of the packages\n doesn't have a patch yet?\n(affected)" }
+    L --> | Yes | X[STATUS: AFFECTED_PATCH_UNAIVAILABLE]
+    L --> | No | W[STATUS: AFFECTED_PATCH_INAPPLICABLE]
+    G --> |Evaluation: false| I[STATUS: PATCHED]
+```
+
+
+
+In Uyuni an audit operation yields an object of type PatchStatus. The PatchStatus type is an enum with four different values:
+
+- AFFECTED_PATCH_INAPPLICABLE  — *Affected, patch available in unassigned channel*
+- AFFECTED_PATCH_APPLICABLE — *Affected, patch available in assigned channel*
+- NOT_AFFECTED — *Not Affected*
+- PATCHED — *Patched*
+- AFFECTED_PATCH_INAPPLICABLE_SUCCESSOR_PRODUCT — *Affected, patch available in a Product Migration target*
+
+With the new algorithm, It is possible to determine two additional patch status namely *AFFECTED_PATCH_UNAVAILABLE* and *AFFECTED_PARTIAL_PATCH_APPLICABLE*. The former indicates that no patch is currently available for the CVE, regardless of the assigned channels, while the latter indicate the a patch is available but won't solve all the issues, meaning the system will still be vulnerable to the Cve, even after applying the patch but safer. The introduction of this new status requires that we rename the other status to better reflect their meaning. The following are the new values of the PatchStatus enum:
+
+- AFFECTED_PATCH_UNAVAILABLE
+- AFFECTED_FULL_PATCH_APPLICABLE
+- AFFECTED_PARTIAL_PATCH_APPLICABLE
+- AFFECTED_PATCH_INAPPLICABLE
+- NOT_AFFECTED
+- PATCHED
+- AFFECTED_PATCH_INAPPLICABLE_SUCCESSOR_PRODUCT
+
+# Drawbacks
+
+# Alternatives
+
+# Unresolved questions
+
+- How to distinguish QA from released patches in non-openSUSE distributions?
+- Can we depend on OVALRepo to provide us with OVAL definitions?
+
```

Comment:
I wonder if we get the querying of data fast enough. First for the existing case, searching for affected systems of 1 CVE number. 
A second case was requested since a long time already. Show all systems with all affected CVE numbers.
Currently this data set it to big to query and present to the user. Maybe this can be done with this style.

During implementation I would try as early as possible to try out the performance of the algorithms.